crates/api/src/module.rs

```diff
@@ -156,12 +159,32 @@ fn read_imports_and_exports(binary: &[u8]) -> Result<(Box<[ImportType]>, Box<[Ex
                     exports.push(ExportType::new(entry.field, r#type));
                 }
             }
+            SectionCode::Custom {
+                kind: CustomSectionKind::Name,
+                ..
+            } => {
+                // Read name section, though ignore invalid data in it.
```

Comment:
Is there a reason to ignore invalid data in the name section?  Is it that we want to be resilient to future breaking changes or is it that displaying the module's name in diagnostics isn't high-priority enough to break when we encounter malformed data?  Granted, for single module scenarios it's not vital, but as things grow to support more complex scenarios it'll be pretty integral for useful diagnostics.

I personally don't think it's unreasonable to get an error for invalid data in custom sections that are used by a runtime.

Reply:
Per spec, the name section shall not be a subject of module validation or error. Also, it says that the name section's purpose "is to attach printable names to definitions in a module" and nothing more.

Reply:
Ah, thanks for the clarification.  Would you mind commenting to that effect for clarity so readers don't think we're just ignoring errors for the fun of it? :)